### PR TITLE
Use mc flag to be able to run with data

### PR DESCRIPTION
**Pull request by chiarazampolli**
Done with @martenole . 
@sawenzel , @nburmaso , this is needed to create AO2D with the pilot beam data. It uses the MC flag to avoid processing MC information if it is not present.

### CODE REVIEW

File: Detectors/AOD/include/AODProducerWorkflow/AODProducerWorkflowSpec.h [hunk 1]
```diff
@@ -191,13 +191,15 @@ typedef boost::unordered_map<Triplet_t, int, TripletHash, TripletEqualTo> Triple
...
  public:
-  AODProducerWorkflowDPL(GID::mask_t src, std::shared_ptr<DataRequest> dataRequest) : mInputSources(src), mDataRequest(dataRequest) {}
+  AODProducerWorkflowDPL(GID::mask_t src, std::shared_ptr<DataRequest> dataRequest, bool useMC = true) : mInputSources(src), mDataRequest(dataRequest), mUseMC(useMC) {}
   ~AODProducerWorkflowDPL() override = default;
   void init(InitContext& ic) final;
   void run(ProcessingContext& pc) final;
   void endOfStream(framework::EndOfStreamContext& ec) final;
 
  private:
+  bool mUseMC = true;
```
Comment (sawenzel) [inline Detectors/AOD/include/AODProducerWorkflow/AODProducerWorkflowSpec.h:201]: I guess we should then directly offer an option for the executable to set this from outside.

Comment (martenole) [reply Detectors/AOD/include/AODProducerWorkflow/AODProducerWorkflowSpec.h:201]: The `aod-producer-workflow` already offers this option, but is was so far only used by the readers: https://github.com/AliceO2Group/AliceO2/blob/36498389afcfff3925b78729dfb6ad990172d38d/Detectors/AOD/src/aod-producer-workflow.cxx#L30

Comment (chiarazampolli) [reply Detectors/AOD/include/AODProducerWorkflow/AODProducerWorkflowSpec.h:201]: As Ole said, the flag is currently an option from the workflow, but we pass it now explicitly to the AOD producer to be able do disable some functionalities (those related to the MC information).

**Comment by shahor02** [comment]
@chiarazampolli could you also change the following:

1) substitute  https://github.com/AliceO2Group/AliceO2/blob/6a462f7b6c8cda85edf4205f40a2e9a2f3e3f511/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L977-L986
by 
```
  std::unique_ptr<o2::steer::MCKinematicsReader> mcReader;
  if (mUseMC) {
    mcReader = std::make_unique<o2::steer::MCKinematicsReader>("collisioncontext.root");
    LOG(DEBUG) << "FOUND " << mcReader->getDigitizationContext()->getEventRecords().size()
	       << " records" << mcReader->getDigitizationContext()->getEventParts().size() << " parts";
  }
  
  std::map<uint64_t, int> bcsMap;
  collectBCs(fddRecPoints, ft0RecPoints, fv0RecPoints, primVertices, mUseMC ? mcReader->getDigitizationContext().getEventRecords() : std::vector<o2::InteractionTimeRecord>{}, bcsMap);
```
2) change https://github.com/AliceO2Group/AliceO2/blob/6a462f7b6c8cda85edf4205f40a2e9a2f3e3f511/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L1074 to 
```
int nMCCollisions = mcReader->getDigitizationContext()->getNCollisions();
const auto& mcRecords = mcReader->getDigitizationContext()->getEventRecords();
const auto& mcParts = mcReader->getDigitizationContext()->getEventParts();
```
3) change in https://github.com/AliceO2Group/AliceO2/blob/6a462f7b6c8cda85edf4205f40a2e9a2f3e3f511/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L1093 to `mcReader->` and in https://github.com/AliceO2Group/AliceO2/blob/6a462f7b6c8cda85edf4205f40a2e9a2f3e3f511/Detectors/AOD/src/AODProducerWorkflowSpec.cxx#L1297 to `(*mcReader)`

`mcReader` to (*mcReader)

**Comment by chiarazampolli** [comment]
Hi @shahor02 . 

The changes you asked for are in (but I committed before compiling :) ).

Chiara

**Comment by sawenzel** [comment]
Is this tested/exercised as part of the full-system test? ... Ah I see #7500. So question answered.

**Comment by davidrohr** [comment]
> Is this tested/exercised as part of the full-system test? ... Ah I see #7500. So question answered.

as of today the AOD creation is included in the full system test.

**Comment by chiarazampolli** [comment]
Hi,
It was also tested with reconstructed data from pilot beam.
Chiara

**Pull request merged**